[2023-12-09T16:04:55.073+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: taskflow.print_result manual__2023-12-09T16:04:52.710865+00:00 [queued]>
[2023-12-09T16:04:55.081+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: taskflow.print_result manual__2023-12-09T16:04:52.710865+00:00 [queued]>
[2023-12-09T16:04:55.081+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 1
[2023-12-09T16:04:55.093+0000] {taskinstance.py:1382} INFO - Executing <Task(PythonOperator): print_result> on 2023-12-09 16:04:52.710865+00:00
[2023-12-09T16:04:55.105+0000] {standard_task_runner.py:57} INFO - Started process 23219 to run task
[2023-12-09T16:04:55.113+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'taskflow', 'print_result', 'manual__2023-12-09T16:04:52.710865+00:00', '--job-id', '89', '--raw', '--subdir', 'DAGS_FOLDER/test_decorator.py', '--cfg-path', '/tmp/tmp_9d792d8']
[2023-12-09T16:04:55.117+0000] {standard_task_runner.py:85} INFO - Job 89: Subtask print_result
[2023-12-09T16:04:55.182+0000] {task_command.py:416} INFO - Running <TaskInstance: taskflow.print_result manual__2023-12-09T16:04:52.710865+00:00 [running]> on host 3de6e37e1218
[2023-12-09T16:04:55.262+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='taskflow' AIRFLOW_CTX_TASK_ID='print_result' AIRFLOW_CTX_EXECUTION_DATE='2023-12-09T16:04:52.710865+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2023-12-09T16:04:52.710865+00:00'
[2023-12-09T16:04:55.265+0000] {python.py:194} INFO - Done. Returned value was: {{ task_instance.xcom_pull(task_ids='print_result', dag_id='adhoc_airflow', key='return_value') }}
[2023-12-09T16:04:55.273+0000] {xcom.py:661} ERROR - Object of type PlainXComArg is not JSON serializable. If you are using pickle instead of JSON for XCom, then you need to enable pickle support for XCom in your *** config or make sure to decorate your object with attr.
[2023-12-09T16:04:55.275+0000] {taskinstance.py:1937} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/json.py", line 91, in default
    return serialize(o)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/serialization/serde.py", line 178, in serialize
    raise TypeError(f"cannot serialize object of type {cls}")
TypeError: cannot serialize object of type <class '***.models.xcom_arg.PlainXComArg'>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/session.py", line 76, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 2479, in xcom_push
    XCom.set(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/session.py", line 76, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/xcom.py", line 244, in set
    value = cls.serialize_value(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/xcom.py", line 659, in serialize_value
    return json.dumps(value, cls=XComEncoder).encode("UTF-8")
  File "/usr/local/lib/python3.8/json/__init__.py", line 234, in dumps
    return cls(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/json.py", line 104, in encode
    return super().encode(o)
  File "/usr/local/lib/python3.8/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/local/lib/python3.8/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/json.py", line 93, in default
    return super().default(o)
  File "/usr/local/lib/python3.8/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type PlainXComArg is not JSON serializable
[2023-12-09T16:04:55.286+0000] {taskinstance.py:1400} INFO - Marking task as FAILED. dag_id=taskflow, task_id=print_result, execution_date=20231209T160452, start_date=20231209T160455, end_date=20231209T160455
[2023-12-09T16:04:55.298+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 89 for task print_result (Object of type PlainXComArg is not JSON serializable; 23219)
[2023-12-09T16:04:55.329+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2023-12-09T16:04:55.348+0000] {taskinstance.py:2778} INFO - 0 downstream tasks scheduled from follow-on schedule check
